plugin/src/index.ts

```diff
@@ -84,15 +106,49 @@ const withShareExtension: ConfigPlugin<{
         height: props?.height,
         preprocessingFile: props?.preprocessingFile,
         googleServicesFile: props?.googleServicesFile,
+        asActionExtension: true,
+      },
+    ],
+    [
+      withShareExtensionInfoPlist,
+      {
+        fonts,
+        activationRules: props?.activationRules,
+        backgroundColor: props?.backgroundColor,
+        height: props?.height,
+        preprocessingFile: props?.preprocessingFile,
+        googleServicesFile: props?.googleServicesFile,
+        asActionExtension: false,
+      },
+    ],
+    [
+      withShareExtensionEntitlements,
+      {
+        asActionExtension: true,
+      },
+    ],
+    [
+      withShareExtensionEntitlements,
+      {
+        asActionExtension: false,
+      },
+    ],
+    [
+      withShareExtensionTarget,
+      {
+        fonts,
+        googleServicesFile: props?.googleServicesFile,
```

Comment:
Refactor config here, there should not be has much duplication. This should be done via js logic